src/hooks/offramp/useOfframpEvents.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useState } from 'preact/compat';
+
+import { EventStatus } from '../../components/GenericEvent';
+import { GenericEvent } from '../../components/GenericEvent';
+
+import { createTransactionEvent } from '../../contexts/events';
+import { useEventsContext } from '../../contexts/events';
+import { useNetwork } from '../../contexts/network';
+
+import { getInputTokenDetailsOrDefault } from '../../constants/tokenConfig';
+import { OfframpingState } from '../../services/offrampingFlow';
+import { OFFRAMPING_PHASE_SECONDS } from '../../pages/progress';
+
+export const useOfframpEvents = () => {
+  const { trackEvent, resetUniqueEvents } = useEventsContext();
+  const { selectedNetwork } = useNetwork();
+  const [_, setEvents] = useState<GenericEvent[]>([]);
+
+  // @todo: why do we need this?
+  const addEvent = (message: string, status: EventStatus) => {
+    setEvents((prevEvents) => [...prevEvents, { value: message, status }]);
+  };
+
+  const trackOfframpingEvent = useCallback(
```

Comment:
Do you think we could move this to `useEventsContext`? Especially since we remove the `addEvent` thingy. This `useOfframpEvents` is a bit intermediary.
If it's too much trouble then no need.

Reply:
I think we should keep the logic in a custom hook and in the Events Context keep only state-related code. The Event Context file is actually quite big right now

Reply:
@gianfra-t I think we should keep the logic in a custom hook and in the Events Context keep only state-related code. The Event Context file is actually quite big right now